app/src/androidTest/java/br/com/catbag/gifreduxsample/idlings/AnvilTestLocker.java

```diff
@@ -31,16 +36,20 @@ public String getName() {
         return AnvilTestLocker.class.getClass().getSimpleName();
     }
 
-    /**
-     * Is important that onStateChanged inside activity control when the Anvil finalize all renders,
-     * eg. putting both flag and the Anvil.render(); inside same thread like GifListActivity does.
-     **/
     @Override
     public boolean isIdleNow() {
-        if (mIdle && mResourceCallback != null) {
-            mResourceCallback.onTransitionToIdle();
+        mHandler.removeCallbacks(mForceIsIdleNow);
+        boolean isIdle = mRenderTimes != 0 && mRenderTimes == mLastRenderTimes;
+        mLastRenderTimes = mRenderTimes;
+
+        if (mResourceCallback != null) {
+            if (isIdle) {
+                mResourceCallback.onTransitionToIdle();
+            } else {
+                mHandler.postDelayed(mForceIsIdleNow, 250);
```

Comment:
Extract a constant from this default interval its better to improve code readability..